server/commands/documentLoader.ts

```diff
@@ -0,0 +1,187 @@
+import invariant from "invariant";
+import { Op } from "sequelize";
+import {
+  NotFoundError,
+  InvalidRequestError,
+  AuthorizationError,
+  AuthenticationError,
+} from "@server/errors";
+import { Collection, Document, Share, User, Team } from "@server/models";
+import { authorize, can } from "@server/policies";
+
+type Props = {
+  id?: string;
+  shareId?: string;
+  user?: User;
+};
+
+type Result = {
+  document: Document;
+  share?: Share;
+  collection: Collection;
+};
+
+export default async function loadDocument({
```

Comment:
Just a direct move of `loadDocument` in `/api/documents.ts` which is already tested very well in that context